community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -472,6 +479,115 @@ def update_topology(lkp: util.Lookup) -> None:
         log.debug("Topology configuration updated. Reconfiguring Slurm.")
         util.scontrol_reconfigure(lkp)
 
+
+def delete_reservations(reservation_map: Dict[str, str]) -> None:
+    for reservation_name in list(reservation_map.keys()):
+      util.run(SLURM_DELETE_RESERVATION.format(reservation_name), timeout=30)
+      del reservation_map[reservation_name]
+
+
+def update_slurm_reservation_maintenance(reservation_name: str, node: str,
+                                         starttime: str, reservation_map: Dict[str, str]) -> None:
+    ## If reservation exists for maintenance, update reservation if time window
+    ## for scheduled maintenance changes.
+    if reservation_map is not None and reservation_name in reservation_map:
```

Comment:
Why would `reservation_map` be `None`, either remove it or update datatype to `Optional[Dict[whatever]]`